annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java

```diff
@@ -671,24 +729,107 @@ private void validateTranscriptionWithFullTextResource(Body body)
+    private void validateVcardAddressBody(Body body) throws ParamValidationException, PropertyValidationException {
 
-	// check mandatory fields
-
-	// check type
+		// check mandatory fields
+    	validateSemanticTagVcardAddressBody(body);
+	
+		// check type
 	if (StringUtils.isBlank(body.getInternalType().toString())
-		|| !BodyInternalTypes.isVcardAddressTagBody(body.getInternalType()))
-	    throw new ParamValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
-		    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD,
-		    new String[] { "tag.body.type", body.getType().toString() });
+			|| !BodyInternalTypes.isVcardAddressTagBody(body.getInternalType()))
+		    throw new ParamValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+			    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD,
+			    new String[] { "tag.body.type", body.getType().toString() });
+    }
+
+	/**
+     * Semantic tagging with vcard:Address type has mandatory fields:
+     * "Address", "streetAddress", "locality", "countryName"
+     * The "type" field should have value "Address"
+     * @param body
+     * @throws ParamValidationException
+	 * @throws PropertyValidationException 
+     */
+    private void validateSemanticTagVcardAddressBody(Body body) throws ParamValidationException, PropertyValidationException {
+		// check mandatory field address
+		if (!(body instanceof VcardAddressBody)) 
+		    throw new ParamValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+				    I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD,
+				    new String[] { "tag.body.class", body.getClass().toString() });
+
+		Address address = ((VcardAddressBody) body).getAddress();
+		
+		// check mandatory field streetAddress
+		if (Strings.isNullOrEmpty(address.getVcardStreetAddress()))
+	        throw new PropertyValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+		        I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD, new String[] { "tag.body.address.streetAddress" });
+
+		// check mandatory field locality
+		if (Strings.isNullOrEmpty(address.getVcardLocality()))
+	        throw new PropertyValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+		        I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD, new String[] { "tag.body.address.locality" });
+
+		// check mandatory field countryName
+		if (Strings.isNullOrEmpty(address.getVcardCountryName()))
+	        throw new PropertyValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
+		        I18nConstants.MESSAGE_MISSING_MANDATORY_FIELD, new String[] { "tag.body.address.countryName" });
+		
+		// check mandatory field type
+		if (body.getType() == null || Strings.isNullOrEmpty(body.getType().get(0)) 
+				|| !body.getType().get(0).equals(WebAnnotationModelKeywords.CLASS_VCARD_ADDRESS))
+	        throw new PropertyValidationException(ParamValidationException.MESSAGE_MISSING_MANDATORY_FIELD,
```

Comment:
When type is different than Address, a different message must be used, Invalid_property_value